mlflow/metrics/metric_definitions.py

```diff
@@ -507,3 +507,32 @@ def _fn(predictions, targets):
         return MetricValue(scores=scores, aggregate_results=standard_aggregations(scores))
 
     return _fn
+
+
+def _bleu_eval_fn(predictions, targets=None, metrics=None):
+    if not _validate_text_data(targets, "bleu", targets_col_specifier) or not _validate_text_data(
+        predictions, "bleu", predictions_col_specifier
+    ):
+        return
+
+    try:
+        bleu = _cached_evaluate_load("bleu")
+    except Exception as e:
+        _logger.warning(f"Failed to load 'bleu' metric (error: {e!r}), skipping metric logging.")
+        return
+
+    result = []
+    for i, (prediction, target) in enumerate(zip(predictions, targets)):
+        if len(target) == 0:
+            _logger.warning(
+                f"Cannot calculate bleu for empty targets: "
+                f"the target is empty on row {i}. Skipping metric logging."
+            )
+            return
```

Comment:
Do we still want to append 0 or something to the result array in the case where one of the target or prediction has a length 0 but we wasted time to compute all the results before that and users still received an error. Maybe we can throw an warning with all the index values where the target or prediction len is 0 so users are aware to discard those values somehow?

Reply:
Good point. I have modified the code so that all invalid entries (where predictions or targets are empty) are accumulated, and I now log a comprehensive warning listing all problematic indices. This should help users easily identify which entries had issues. I also append `0`s for these entries to the results array to ensure the output has consistent length and is easy to interpret. 
Let me know if this solution addresses your concerns or if you have any further suggestions.